gitlab/utils.py

```diff
@@ -47,6 +49,39 @@ def response_content(
     return None
 
 
+def _transform_types(
+    data: Dict[str, Any], custom_types: dict, transform_files: Optional[bool] = True
+) -> Tuple[dict, dict]:
```

Comment:
What do you think about using a dataclass for the return value?

```
@dataclasses.dataclass
class FileInfoOrSomeBetterName:
   filename: str
   data: bytes

@dataclasses.dataclass
class SomeSuperAwesomeName:
   data: Dict[str, Any]
   files: Dict[str, FileInfoOrSomeBetterName]
```

Not required.

Reply:
This was mostly a copy/paste refactor to extract the duplicate code to a common helper, but if I do more changes I'd like to get rid of `files` completely and handle everything in the payload without this extremely roundabout way we have now (args -> FileAttribute -> files -> data).

I'll give that a shot here, but if it gets messy I'll add the dataclasses as suggested :)